### PR TITLE
Fix problem which lead redisson plugin to throw unnecessary NullPointerException

### PR DESCRIPTION
**Pull request by Ricehomesky**
Change RedisConnectionInstrumentation and RedisConnectionMethodInterceptor to v2 interface, to solve the problem in this issue #693 .

### CODE REVIEW

File: apm-sniffer/apm-sdk-plugin/redisson-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/redisson/v3/RedisConnectionMethodInterceptor.java [hunk 1]
```diff
@@ -154,8 +159,8 @@ private Optional<String> parseOperation(String cmd) {
 
     private String showBatchCommands(CommandsData commandsData) {
         return commandsData.getCommands()
-                           .stream()
-                           .map(data -> data.getCommand().getName())
-                           .collect(Collectors.joining(";"));
+                        .stream()
+                        .map(data -> data.getCommand().getName())
+                        .collect(Collectors.joining(";"));
```
Comment (wu-sheng) [inline apm-sniffer/apm-sdk-plugin/redisson-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/redisson/v3/RedisConnectionMethodInterceptor.java:164]: These seem unnecessary.

File: apm-sniffer/apm-sdk-plugin/redisson-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/redisson/v3/RedisConnectionMethodInterceptor.java [hunk 2]
```diff
@@ -93,17 +96,19 @@ public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allAr
...
         return ret;
     }
 
     @Override
-    public void handleMethodException(EnhancedInstance objInst, Method method, Object[] allArguments,
-        Class<?>[] argumentsTypes, Throwable t) {
-        AbstractSpan span = ContextManager.activeSpan();
-        span.log(t);
+    public void handleMethodException(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes, Throwable t, MethodInvocationContext context) {
+        if (ContextManager.isActive()) {
```
Comment (wu-sheng) [inline apm-sniffer/apm-sdk-plugin/redisson-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/redisson/v3/RedisConnectionMethodInterceptor.java:108]: You should use flag here as well.

Comment (Ricehomesky) [reply apm-sniffer/apm-sdk-plugin/redisson-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/redisson/v3/RedisConnectionMethodInterceptor.java]: ok, i fixed these two problems right now

File: apm-sniffer/apm-sdk-plugin/redisson-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/redisson/v3/RedisConnectionMethodInterceptor.java [hunk 3]
```diff
@@ -81,6 +83,7 @@ public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allAr
             }
         }
 
+        context.setContext(STOP_SPAN_FLAG);
         AbstractSpan span = ContextManager.createExitSpan(operationName, peer);
```
Comment (wu-sheng) [inline apm-sniffer/apm-sdk-plugin/redisson-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/redisson/v3/RedisConnectionMethodInterceptor.java:87]:
```suggestion
        AbstractSpan span = ContextManager.createExitSpan(operationName, peer);
        context.setContext(STOP_SPAN_FLAG);
```

Put the flag after Created, this should be safer.

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**